### PR TITLE
chore: Improved profiling

### PR DESCRIPTION
**Pull request by fredizzimo**
## What kind of change does this PR introduce?
* Add support for profiling async tasks
* Add support for plotting graphs when profiling
* Add support for named frames (shows the duration of a re-occuring task) with a name
* Add more profiling events that helps especially with viewing the event flow

## Did this PR introduce a breaking change? 
_A breaking change includes anything that breaks backwards compatibility either at compile or run time._
- No

### CODE REVIEW

File: src/renderer/cursor_renderer/mod.rs [hunk 1]
```diff
@@ -422,6 +422,7 @@ impl CursorRenderer {
         if !animating {
             self.previous_editor_mode = current_mode.clone();
         }
+        tracy_plot!("cursor animating", animating as u8 as f64);
```
Comment (MultisampledNight) [inline src/renderer/cursor_renderer/mod.rs:425]: What is the thought behind the double cast?

Comment (fredizzimo) [reply src/renderer/cursor_renderer/mod.rs:425]: The plots only takes f64, and animating as a boolean, which is not directly convertible to floats

Comment (MultisampledNight) [reply src/renderer/cursor_renderer/mod.rs:425]: Oh. That makes sense, thanks!

**Comment by MultisampledNight** [comment]
Thank you! :steam_locomotive: Chooooooooo choooooo!

**Pull request merged**